File: lecture3/src/main/java/at/technikumwien/maps/data/local/DrinkingFountainDao.java
package at.technikumwien.maps.data.local;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface DrinkingFountainDao {

    @Query("SELECT * FROM drinking_fountains")
    List<DrinkingFountain> getAllDrinkingFountains();

    // @Query("SELECT fullname FROM drinking_fountains")
    // List<String> getDrinkingFountainNames();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(DrinkingFountain... drinkingFountains);

    @Query("DELETE FROM drinking_fountains")
    void deleteAll();
}

File: exercise3/src/main/java/at/technikumwien/maps/data/local/DrinkingFountainDao.kt
package at.technikumwien.maps.data.local

import androidx.lifecycle.LiveData
import androidx.room.*

import at.technikumwien.maps.data.model.DrinkingFountain
import at.technikumwien.maps.data.model.DrinkingFountain.Companion.TABLE_NAME

@Dao
interface DrinkingFountainDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(drinkingFountains: List<DrinkingFountain>)

    @Query("DELETE FROM $TABLE_NAME")
    suspend fun deleteAll()

    @Transaction
    suspend fun refresh(drinkingFountains: List<DrinkingFountain>) {
        deleteAll()
        insert(drinkingFountains)
    }

    @Query("SELECT * FROM $TABLE_NAME")
    fun loadAll(): LiveData<List<DrinkingFountain>>

}
